File: leetcode_notes/leetcode/0206.反转链表/0206-反转链表.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    // 206
    // Reference: cspiration
    public ListNode reverseList(ListNode head) {
        // Time: O(n) Space: O(1)
        if(head == null || head.next == null) return head;
        ListNode pre = null;
        while(head != null) {
            ListNode temp = head.next;
            head.next = pre;
            pre = head;
            head = temp;
        }
        return pre;
    }
}

File: leetcode_notes/leetcode_cn/0206.反转链表/0206-反转链表.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(head==NULL || head->next==NULL) return head;  
          
        ListNode* pre = head;  
        ListNode* p = head->next;  
        pre->next = NULL;  
        ListNode* nxt;  
        while(p!=NULL) {  
            nxt = p->next;  
            p->next = pre;  
            pre = p;  
            p = nxt;  
        }  
        return pre;  
    }  
};